File: feature/copy/export/src/main/java/com/android/copy/export/permission/ManifestRegisterException.java
package com.android.open9527.per;

/**
 * @author open_9527
 * Create at 2021/5/19
 **/
final class ManifestRegisterException extends RuntimeException {

    ManifestRegisterException() {
        // 清单文件中没有注册任何权限
        super("No permissions are registered in the manifest file");
    }

    ManifestRegisterException(String permission) {
        // 申请的危险权限没有在清单文件中注册
        super(permission + ": Permissions are not registered in the manifest file");
    }
}

File: lib_kt/permissions/src/main/java/com/android/open9527/permissions/ManifestRegisterException.kt
package com.android.open9527.permissions

/**
 * @author open_9527
 * Create at 2021/5/19
 */
internal class ManifestRegisterException : RuntimeException {
    constructor() : super("No permissions are registered in the manifest file") {
        // 清单文件中没有注册任何权限
    }

    constructor(permission: String) : super("$permission: Permissions are not registered in the manifest file") {
        // 申请的危险权限没有在清单文件中注册
    }
}